### PR TITLE
Max6956 support added

### PR DESCRIPTION
**Pull request by looping40**
# What does this implement/fix?

Quick description and explanation of changes

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2138

## Test Environment

- [x] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs/pull/2138).

### CODE REVIEW

File: esphome/components/max6956/__init__.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+@automation.register_action(
+    "max6956.set_brightness_global",
+    SetCurrentGlobalAction,
+    cv.Schema(
+        {
+            cv.Required(CONF_ID): cv.use_id(MAX6956),
+            cv.Required(CONF_BRIGHTNESS_GLOBAL): cv.templatable(cv.percentage),
+        }
+    ),
+)
```
Comment (GregJArnold) [inline esphome/components/max6956/__init__.py:117]: Should there also be an automation to set the brightness mode to global or individual?

File: esphome/components/max6956/automation.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+namespace esphome {
+namespace max6956 {
+
+template<typename... Ts> class SetCurrentGlobalAction : public Action<Ts...> {
+ public:
+  SetCurrentGlobalAction(MAX6956 *max6956) : max6956_(max6956) {}
+
+  TEMPLATABLE_VALUE(float, brightness_global)
+
+  void play(Ts... x) override { this->max6956_->set_brightness_global(16 * this->brightness_global_.value(x...)); }
```
Comment (GregJArnold) [inline esphome/components/max6956/automation.h:16]: I could be mistaken, but I think that this just sets the class variable, but doesn't actually write it out to the I²C bus.

**Review by GregJArnold — CHANGES_REQUESTED** [review]
I think it looks good, except for the perhaps-missing action, and the one comment about setting the global current settings.

File: esphome/components/max6956/__init__.py (hunk 1 above)
Comment (looping40) [reply esphome/components/max6956/__init__.py:117]: I implemented this modification on my local repository, cloned https://github.com/looping40/esphome, but I can't commit : it seems I should not be on the dev branch.
How can I solve this ?

**Comment by looping40** [comment]
I created the PR using a Dev branch so I could not commit the changes required by the review.
The only way to fix this seems  to close this PR and create a new one, on a dedicated branch:
https://github.com/esphome/esphome/pull/3764

**Pull request closed without merging**